aten/src/ATen/templates/TensorBody.h

```diff
@@ -394,13 +447,178 @@ class CAFFE2_API Tensor {
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  /// Gets the gradient function of the `Variable`. If this is a leaf variable,
+  /// the pointer returned will be null.
+  ///
+  /// For View Variables:
+  /// Gets the up-to-date grad_fn. If the shared data or base was modified, we
+  /// re-create the grad_fn to express the up-to-date view relationship between
+  /// this and the base Variable.
+  const std::shared_ptr<torch::autograd::Node>& grad_fn() const;
+
+  /// Gets the raw gradient function pointer, whatever it currently is.
+  torch::autograd::Node* grad_fn_unsafe() const;
+
+  /// Attempts to get a pointer to the gradient accumulator of the `Variable`,
+  /// if it still exists. If the gradient accumulator function has been
+  /// destroyed, returns a `nullptr`.
+  std::shared_ptr<torch::autograd::Node> try_get_grad_accumulator() const;
+
+  /// Gets the gradient accumulator of the `Variable` if it has one, or else
+  /// create one on the fly and return it.
+  std::shared_ptr<torch::autograd::Node> grad_accumulator() const;
+
+  /// Returns the "canonical" gradient edge of this `Variable`, i.e. either the
+  /// gradient function if this is an interior `Variable`, or the gradient
+  /// accumulator otherwise. If the `Variable` is interior, the returned `Edge`
+  /// will store the input index of the `Node` to which this variable is
+  /// connected in its `input_nr` field. For leaves, the `input_nr` is always
+  /// zero. Note that `set_gradient_edge` and `gradient_edge` are not
+  /// symmetric. You must use `set_gradient_edge` to set the `grad_fn` and
+  /// `set_grad_accumulator` to set the accumulator.
+  torch::autograd::Edge gradient_edge() const {
+    // If grad_fn is null (as is the case for a leaf node), we instead
+    // interpret the gradient function to be a gradient accumulator, which will
+    // accumulate its inputs into the grad property of the variable. These
+    // nodes get suppressed in some situations, see "suppress gradient
+    // accumulation" below. Note that only variables which have `requires_grad =
+    // True` can have gradient accumulators.
+    if (const auto& gradient = grad_fn()) {
+      return torch::autograd::Edge(gradient, output_nr());
+    } else {
+      return torch::autograd::Edge(grad_accumulator(), 0);
+    }
+  }
+
+  /// Set the gradient edge -- i.e. `grad_fn` and `input_nr` -- of the
+  /// `Variable`.
+  /// NOTE: This will always set the `grad_fn`, even if this is a leaf variable,
+  /// and never the `grad_accumulator`. For the latter, use
+  /// `set_grad_accumulator`. This allows late construction of an interior
+  /// `Variable`.
+  void set_gradient_edge(torch::autograd::Edge edge) const noexcept;
+
+  // Versions
+  //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  /// Increments the version count of this `Variable`.
+  void bump_version() const noexcept {
+    impl_->bump_version();
+  }
```

Comment:
I guess this might be a hypothetically useful operation for users, but I wouldn't consider it as part of the public API.